### DISCUSSION TITLE
Suggestion: support XPath via `document.evaluate`

### DISCUSSION THREAD

**Discussion by elazarcoh** [gnat/surreal#47] [Ideas]
This is the first time I needed something like this, so I don't know how useful it is in general.

I needed to "hack" my way to modify the style of an element, in a 3rdparty framework. I couldn't use any available method in surreal to get the correct element (again, because it's a 3rdparty framework, and I had no control over any of the attributes/classes).
So instead, I used an XPath expression with `document.evaluate` to find the correct element and then applied `me` on it to get all the other benefits of surreal.

So, the suggestion is to support XPath expressions in the `me` and `any` methods.

**Comment by gnat** [comment]
This is a good idea!

If anyone wants to take an initial crack at it, you're more than welcome to.
